### ISSUE TITLE
Meta definitions for plugin-imported modules

### ISSUE THREAD

**Issue by olee** [jspm/jspm-cli#2209]
Hi, I'm trying to define dependencies for css files imported through the css plugin.
````js
System.config({
  ...
  meta: {
    "./style.css!": {
      "deps": [
        "angular-material/angular-material.min.css!"
      ]
    },
  },
````
I found out after some debugging, that the module name `./style.css!.js` gets expanded into `http://localhost:3000/style.css!.js` which is incorrect, making the dependencies not work.
It seems the `.js` is appendend because of `defaultJSExtensions: true` setting, which I cannot turn off however.
Is there any way to make the meta definition work?
I think this is actually a bug and there should probably be some way to fix this.

**Comment by aluanhaddad** [comment]
Hello.
Have you tried the following?
```js
SystemJS.config({
  meta: {
    "*.css" {
      defaultExtension: false
    }
  }
});
```

**Comment by olee** [comment]
Doesn't work.
The problem is this code in SystemJS:
````js
        else if (c == 'meta') {
          // meta can go through global map, with defaultJSExtensions adding
          var resolved = coreResolve.call(loader, p);
          if (loader.defaultJSExtensions && resolved.substr(resolved.length - 3, 3) != '.js' && !isPlain(resolved))
            resolved += '.js';
          extend(loader[c][resolved] = loader[c][resolved] || {}, v[p]);
        }
````

**Comment by aluanhaddad** [comment]
What version of jspm and SystemJS are you using? Can put together a minimal repro?

**Comment by olee** [comment]
I can try to when I get the time.
However it should be quite easy to get a (not) working sample by just creating a new jspm project, adding css plugin and trying to set up dependencies for imported css files.

**Comment by aluanhaddad** [comment]
I have done that and it works. Perhaps share your version and configuration?

**Comment by danwellman** [comment]
I'm seeing this issue also. I'm using JSPM version 0.16.53, which installed SystemJS version 0.19.46

After installing the [sass plugin](https://github.com/dougludlow/plugin-sass), the config.js file was updated with a mapping for the plugin: `"sass": "github:dougludlow/plugin-sass@0.6.0",`

I've also added config to use the plugin for all scss files using the meta object:

    meta: {
      "*.scss": { "defaultExtension": false, "loader": "sass" }
    },

Unfortunately, when I try to load an scss file, e.g. `import 'scss/styles.scss';` I see a message in the browser console that says `scss.scss.js could not be found`

Is there no way to override the defaultJSExtensions config that JSPM adds?
